packages/web-components/@groupby/elements-products/src/products-base.ts

```diff
@@ -40,6 +45,34 @@ export default class ProductsBase extends Base {
     }
   }
 
+  /**
+   * Requests initial data for the component.
+   *
+   * @param name The name of the initial cached data to return.
+   * @param group The name of the event group that the component belongs to.
+   * @param returnEvent The name of the event under which to dispatch the cached data.
+   *
+   */
+  requestInitialData(name: string, group: string, returnEvent: string): void {
+    const payload: CacheRequestPayload = {
+      name,
+      group,
+      returnEvent,
+    };
+    this.dispatchElementsEvent<CacheRequestPayload>(CACHE_REQUEST, payload);
+  }
+
+  /**
+   * Generates a string intended to be used as the name of the return event in
+   * cache requests for the component.
+   *
+   * @param componentName The name of the component for which to generate the return event name.
+   * @param componentId A random string that acts as a unique identifier for the component.
+   */
+  getCacheResponseEventName(componentName: string, componentId: string): string {
```

Comment:
This can be turned into a getter. Please refer to the autocomplete component for an example.

Reply:
@jmbtutor I was not able to make this a getter, it threw an error about `A 'get' accessor cannot have parameters.` I did follow your suggestion to make both the protected cacheResponseEventName in products-search and products-sayt into getters. I'll submit this next round momentarily and see what you think. I think there might be a cleaner way to do all this but Im interested in your next review

Reply:
done